e2e-tests/tests.rs

```diff
@@ -0,0 +1,164 @@
+use e2e_tests::ethereum::EthereumClient;
+use e2e_tests::localstack::LocalStack;
+use e2e_tests::sharp::SharpClient;
+use e2e_tests::starknet_client::StarknetClient;
+use e2e_tests::{
+    get_env_var_or_panic, get_mongo_db_client, mock_proving_job_endpoint_output, mock_starknet_get_nonce,
+    mock_starknet_get_state_update, put_job_data_in_db_da, put_job_data_in_db_snos, put_job_data_in_db_update_state,
+    MongoDbServer, Orchestrator,
+};
+use mongodb::bson::doc;
+use orchestrator::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use orchestrator::queue::job_queue::WorkerTriggerType;
+use std::time::Duration;
+use tokio::time::sleep;
+
+extern crate e2e_tests;
+
+#[tokio::test]
+async fn test_orchestrator_workflow() {
+    // Fetching the env vars from the test env file because setting up of the environment
+    // requires all these variables.
+    dotenvy::from_filename(".env.test").expect("Failed to load the .env file");
+
+    let (
+        mongo_db_instance,
+        mut starknet_client_mock,
+        _ethereum_client_mock,
+        mut sharp_client_mock,
+        env_vec,
+        l2_block_number,
+    ) = setup_for_test().await.unwrap();
+
+    // Step 1 : SNOS job runs =========================================
+    // TODO : Update the code with actual SNOS implementation
+    // Updates the job in the db
+    put_job_data_in_db_snos(&mongo_db_instance, l2_block_number.clone()).await;
+
+    // Step 2: Proving Job ============================================
+    // Mocking the endpoint
+    mock_proving_job_endpoint_output(&mut sharp_client_mock).await;
+
+    // Step 3: DA job =================================================
+    // mocking get_block_call from starknet client
+
+    // Adding a mock da job so that worker does not create 60k+ jobs
+    put_job_data_in_db_da(&mongo_db_instance, l2_block_number.clone()).await;
+    mock_starknet_get_state_update(&mut starknet_client_mock, l2_block_number.clone()).await;
+    mock_starknet_get_nonce(&mut starknet_client_mock, l2_block_number.clone()).await;
+
+    // Step 4: State Update job =======================================
+    put_job_data_in_db_update_state(&mongo_db_instance, l2_block_number).await;
+
+    println!("✅ Orchestrator setup completed.");
+
+    // Run orchestrator
+    let mut orchestrator = Orchestrator::run(env_vec);
+    orchestrator.wait_till_started().await;
+
+    // TODO : need to make this dynamic
+    sleep(Duration::from_secs(900)).await;
+
+    // Adding a case here to check for required state of the orchestrator to end the test.
+    let l2_block_for_testing = get_env_var_or_panic("L2_BLOCK_NUMBER_FOR_TEST");
```

Comment:
why does this need to be an env var?

Reply:
So my intention to keep it there was in case we need to test some other block pie in e2e test.

Reply:
i see, but then we can change the number in the test case itself. the env var feels sort of incorrect here